exercises/grade-school/canonical-data.json

```
@@ -1,113 +1,234 @@
+        "Ensure the implementation does not prevent addition of other students after preventing the error case"
+      ],
+      "property": "roster",
+      "input": {
+        "students": [["Blair", 2], ["James", 2], ["James", 2], ["Paul", 2]]
+      },
+      "expected": ["Blair", "James", "Paul"]
+    },
+    {
+      "uuid": "e70d5d8f-43a9-41fd-94a4-1ea0fa338056",
+      "description": "Adding students in multiple grades",
+      "property": "add",
+      "input": {
+        "students": [["Chelsea", 3], ["Logan", 7]]
+      },
+      "expected": [true, true]
+    },
+    {
+      "uuid": "75a51579-d1d7-407c-a2f8-2166e984e8ab",
+      "description": "Students in multiple grades are added to the roster",
+      "property": "roster",
+      "input": {
+        "students": [["Chelsea", 3], ["Logan", 7]]
+      },
+      "expected": ["Chelsea", "Logan"]
+    },
+    {
+      "uuid": "7df542f1-57ce-433c-b249-ff77028ec479",
+      "description": "Cannot add same student to multiple grades in the roster",
+      "property": "add",
+      "input": {
+        "students": [["Blair", 2], ["James", 2], ["James", 3], ["Paul", 3]]
+      },
+      "expected": [true, true, false, true]
+    },
+    {
+      "uuid": "6a03b61e-1211-4783-a3cc-fc7f773fba3f",
+      "reimplements": "c125dab7-2a53-492f-a99a-56ad511940d8",
+      "comments": [
+        "Reimplemented to be logically consistent"
+      ],
+      "description": "A student cannot be added to more than one grade in the sorted roster",
+      "property": "roster",
+      "input": {
+        "students": [["Aimee", 2], ["Aimee", 1]]
+      },
+      "expected": ["Aimee"]
+    },
+    {
+      "uuid": "c7ec1c5e-9ab7-4d3b-be5c-29f2f7a237c5",
+      "description": "Student not added to multiple grades in the roster",
+      "reimplements": "6a03b61e-1211-4783-a3cc-fc7f773fba3f",
+      "comments": [
+        "Ensure the implementation does not affect students already added",
+        "Ensure the implementation does not prevent addition of other students after preventing the error case"
+      ],
+      "property": "roster",
+      "input": {
+        "students": [["Blair", 2], ["James", 2], ["James", 3], ["Paul", 3]]
+      },
+      "expected": ["Blair", "James", "Paul"]
+    },
+    {
+      "uuid": "d9af4f19-1ba1-48e7-94d0-dabda4e5aba6",
+      "description": "Students are sorted by grades in the roster",
+      "property": "roster",
+      "input": {
+        "students": [["Jim", 3], ["Peter", 2], ["Anna", 1]]
+      },
+      "expected": ["Anna", "Peter", "Jim"]
+    },
+    {
+      "uuid": "d9fb5bea-f5aa-4524-9d61-c158d8906807",
+      "description": "Students are sorted by name in the roster",
+      "property": "roster",
+      "input": {
+        "students": [["Peter", 2], ["Zoe", 2], ["Alex", 2]]
+      },
+      "expected": ["Alex", "Peter", "Zoe"]
+    },
+    {
+      "uuid": "180a8ff9-5b94-43fc-9db1-d46b4a8c93b6",
+      "description": "Students are sorted by grades and names in the roster",
```

Maybe rewrite slightly to make the sorting order clearer?

```suggestion
      "description": "Students are sorted by grade and then by name in the roster",
```